source/textUtils.py

```diff
@@ -417,6 +417,182 @@ def encodedToStrOffsets(
+						normalizedPart = unicodedata.normalize(self.normalizationForm, originPart)
+						if (
+							originPart == normalizedPart
+							or not normalizedBuffer.startswith(normalizedPart)
+						):
+							continue
+						originPartLen = len(originPart)
+						originBuffer = originBuffer[originPartLen:]
+						normalizedPartLen = len(normalizedPart)
+						normalizedBuffer = normalizedBuffer[normalizedPartLen:]
+						break
+					# Map the original indices to the normalized indices.
+					# originMultiplier is used to multiply indices in origin
+					# when a character takes more space in origin than in normalized.
+					# This is applicable when normalizing letter+modifier compositions to one character.
+					originMultiplier = min(originPartLen / normalizedPartLen, 1)
+					# normalizedMultiplier is used to multiply indices in normalized
+					# when a character takes more space in normalized than in origin.
+					# This is applicable when normalizing one character ligeatures
+					# into their two corresponding letters.
+					normalizedMultiplier = min(normalizedPartLen / originPartLen, 1)
+					for i in range(max(originPartLen, normalizedPartLen)):
+						tempOrigin = iOrigin + int(i * originMultiplier)
+						tempNormalized = iNormalized + int(i * normalizedMultiplier)
+						originC = originPart[i] if i < originPartLen else None
+						if originC:
+							# If normalization results in the same characters
+							# but they have moved in the string, for example when normalizing the order of modifiers
+							# on ancient hebrew consonants, the normalized index should be based on
+							# the position of the origin character in normalized.
+							normalizedIndex = normalizedPart.find(originC)
+							if normalizedIndex != -1:
+								tempNormalized = iNormalized + normalizedIndex
+						normalizedC = normalizedPart[i] if i < normalizedPartLen else None
+						if normalizedC:
+							# The origin index should be based on the position
+							# of the normalized character in origin.
+							originIndex = originPart.find(normalizedC)
+							if originIndex != -1:
+								tempOrigin = iOrigin + originIndex
+						originToNormalizedDict[tempOrigin].append(tempNormalized)
+						normalizedToOriginDict[tempNormalized].append(tempOrigin)
+					iOrigin += originPartLen
+					iNormalized += normalizedPartLen
+				originPending = normalizedPending = False
+				if char[0] == " ":
+					# Map indices directly for unchanged characters.
+					originToNormalizedDict[iOrigin].append(iNormalized)
+					normalizedToOriginDict[iNormalized].append(iOrigin)
+					iOrigin += 1
+					iNormalized += 1
+		# Finalize the mapping by selecting the minimum index for each original position.
+		originResult = tuple(map(min, originToNormalizedDict.values()))
+		assert len(originResult) == len(self.decoded)
+		normalizedResult = tuple(map(min, normalizedToOriginDict.values()))
+		assert len(normalizedResult) == len(self.encoded)
+		return tuple((

```

Comment:
Isn't this a tuple as is?

```suggestion
		return (
```

Reply:
I think you are right. I personally notice that I prefer the more explicit syntax because only brackets can also produce a generator if there is a list comprehension between the brackets. But as far as I'm concerned, the final judgment is up to you.
@Coderabbitai what do you think about the suggestion made here?